web/client/components/TOC/DefaultLayer.jsx

```diff
@@ -118,12 +119,15 @@ class DefaultLayer extends React.Component {
             <div key="legend" position="collapsible" className="collapsible-toc">
                 <Grid fluid>
                     {this.props.showFullTitleOnExpand ? <Row><Col xs={12} className="toc-full-title">{this.getTitle(this.props.node)}</Col></Row> : null}
-                    {this.props.activateLegendTool ?
+                    {this.props.activateLegendTool && this.props.node.type === 'wms' &&
                         <Row>
                             <Col xs={12}>
                                 <WMSLegend node={this.props.node} currentZoomLvl={this.props.currentZoomLvl} scales={this.props.scales} language={this.props.language} {...this.props.legendOptions} />
                             </Col>
-                        </Row> : null}
+                        </Row>}
+                    {this.props.activateLegendTool && ['wfs', 'vector'].includes(this.props.node.type) &&
```

Comment:
based on the previous comment
```
['wfs', 'vector', '3dtiles']
```

Reply:
as discussed we can skip the `3dtiles` for the moment,
@belom88 thanks for checking